models/demos/tg/llama3_70b/demo/demo.py

```diff
@@ -255,14 +307,21 @@ def run_decode(
         input_tokens = tokens[:, prev_pos:cur_pos]
         logits = model.forward(input_tokens, prev_pos)
 
+        if cur_pos == min_prompt_len:
+            total_time_to_first_token_end = time()
+
         next_logits = logits[:, -1, :]  # batch, vocab of last token
         next_token = sampling_func(next_logits)
 
         tokens, eos_reached, prev_pos = prepare_next_input(
             tokenizer, tokens, input_text_mask, finished_mask, prompt_lens, cur_pos, next_token
         )
         latencies.append(time() - start)
-
+        # If compilation run for 2 iterations (prefill+decode)
+        if compilation and min_prompt_len + 2 == cur_pos:
```

Comment:
Wouldn't it be enough to just call model.forward and prepare_next_input in a loop for warmup? Or even push random inputs? I think it would be cleaner if we wouldn't re-use run_decode and then continue/break dependent on compilation and what position we are in.

Reply:
I believe doing everything the same what run_decode already does in warmup step would make code more complex and when we can reuse run_decode I believe we shouldn't add more code to there. I agree that if statements for compilation make run_decode less clean but I wouldn't say it's less understandable or more complex. Reusing it for compilation step clearly shows that we are doing exactly the same thing as in forward pass later.